cli/config/memorystore/store.go

```diff
@@ -0,0 +1,103 @@
+package memorystore
+
+import (
+	"errors"
+	"maps"
+
+	"github.com/docker/cli/cli/config/credentials"
+	"github.com/docker/cli/cli/config/types"
+)
+
+var errValueNotFound = errors.New("value not found")
+
+func IsErrValueNotFound(err error) bool {
+	return errors.Is(err, errValueNotFound)
+}
+
+type memoryStore struct {
+	memoryCredentials map[string]types.AuthConfig
+	fallbackStore     credentials.Store
+}
+
+func (e *memoryStore) Erase(serverAddress string) error {
+	delete(e.memoryCredentials, serverAddress)
+
+	if e.fallbackStore != nil {
+		if err := e.fallbackStore.Erase(serverAddress); err != nil {
```

Comment:
Wondering if this should handle (and ignore) not-found errors from the fallback store. It's likely the fallback store won't have the creds, and in that case, a logout probably shouldn't produce an error (assuming this returns a "not found" error).